neuraxle/steps/logic.py

```diff
@@ -0,0 +1,134 @@
+import numpy as np
+
+from neuraxle.base import BaseStep, ExecutionContext, NonFittableMixin
+from neuraxle.data_container import DataContainer, ListDataContainer
+from neuraxle.hyperparams.distributions import HyperparameterDistribution, FixedHyperparameter
+from neuraxle.hyperparams.space import HyperparameterSamples, HyperparameterSpace
+from neuraxle.pipeline import Pipeline
+from neuraxle.steps.column_transformer import ColumnChooserTupleList, ColumnsSelectorND
+
+
+class FilterColumns(NonFittableMixin, Pipeline):
+    def __init__(self, conditions: ColumnChooserTupleList, n_dimension: int = 3):
```

Comment:
I have a very hard time understanding what this class does exactly and why it exists. 

Dimensions, types, and behavior must be documented before merging.